marshal_test.go

```diff
@@ -3976,3 +3977,59 @@ func TestGithubIssue431(t *testing.T) {
 		t.Errorf("UnmarshalTOML should not have been called")
 	}
 }
+
+type durationString struct {
+	time.Duration
+}
+
+func (d *durationString) UnmarshalTOML(v interface{}) error {
+	d.Duration = 10 * time.Second
+	return nil
+}
+
+type config437Error struct {
+}
+
+func (e *config437Error) UnmarshalTOML(v interface{}) error {
+	return errors.New("expected")
+}
+
+type config437 struct {
+	HTTP struct {
+		PingTimeout durationString `toml:"PingTimeout"`
+		ErrorField  config437Error
+	} `toml:"HTTP"`
+}
+
+func TestGithubIssue437(t *testing.T) {
+	src := `
+[HTTP]
+PingTimeout = "32m"
+`
+	cfg := &config437{}
+	cfg.HTTP.PingTimeout = durationString{time.Second}
+
+	r := strings.NewReader(src)
+	err := NewDecoder(r).Decode(cfg)
+	if err != nil {
+		t.Fatalf("unexpected errors %s", err)
+	}
+	expected := durationString{10 * time.Second}
+	if cfg.HTTP.PingTimeout != expected {
```

Comment:
@pelletier I recommend you to take a look at [`github.com/stretchr/testify`](https://github.com/stretchr/testify) package.

It contains a set of handy tools for unit tests - [`require`](https://pkg.go.dev/github.com/stretchr/testify@v1.6.1/require?tab=doc) and [`assert`](https://pkg.go.dev/github.com/stretchr/testify@v1.6.1/assert?tab=doc)

```go
// assert.NoError is also ok, but require will stop test if condition is failed
require.NoError(t, err)
require.EqualError(t, err, "some error message")
require.Equal(t, expected, cfg.HTTP.PingTimeout)
```

Reply:
Thanks!